### PR TITLE
Introduce concept of "system SMS config"

### PR DESCRIPTION
**Pull request by sethvargo**
First part of GH-432.

There's currently no way to create or delegate SMS configs - that will be a follow-up PR. All of the UI is guarded by the SMS config being delegated though.

**Release Note**

```release-note
Introduce scaffolding for system SMS configs
```

### CODE REVIEW

File: cmd/server/assets/header.html [hunk 1]
```diff
@@ -497,6 +497,14 @@ <h6 class="dropdown-header">Actions</h6>
 </div>
 {{end}}
 
+{{define "errorable"}}
```
Comment (sethvargo) [inline cmd/server/assets/header.html:500]: I'll go back and update our existing forms to use this someday.

File: cmd/server/assets/realmadmin/_form_abuse_prevention.html [hunk 2]
```diff
@@ -19,13 +19,14 @@
   </p>
 
   <div class="form-group form-check">
-    <input type="checkbox" name="abuse_prevention_enabled" id="abuse-prevention-enabled" class="form-check-input" value="1" {{if $realm.AbusePreventionEnabled}} checked{{end}}>
+    <input type="checkbox" name="abuse_prevention_enabled" id="abuse-prevention-enabled" class="form-check-input" value="1" {{if $realm.AbusePreventionEnabled}} checked{{end}}
```
Comment (sethvargo) [inline cmd/server/assets/realmadmin/_form_abuse_prevention.html:22]: Turns out there's built-in Bootstrap for this...

File: pkg/controller/realmadmin/settings.go [hunk 3]
```diff
@@ -105,14 +106,6 @@ func (c *Controller) HandleSettings() http.Handler {
 			return
 		}
 
-		// SMS config is all or nothing
-		if (form.TwilioAccountSid != "" || form.TwilioAuthToken != "" || form.TwilioFromNumber != "") &&
```
Comment (sethvargo) [inline pkg/controller/realmadmin/settings.go:109]: Moved this into the model in preparation for having a system admin page to create Twilio credentials.

File: pkg/controller/realmadmin/settings.go [hunk 4]
```diff
@@ -131,6 +124,11 @@ func (c *Controller) HandleSettings() http.Handler {
 			realm.SMSTextTemplate = form.SMSTextTemplate
 		}
 
+		// SMS
```
Comment (mikehelmick) [inline pkg/controller/realmadmin/settings.go:127]: I think we should refactor this controller to have a separate action per tab on the form, can file as a followup issue.

**Review by mikehelmick — APPROVED** [review]

**Pull request merged**